packages/next/src/build/next-config-ts/utils.ts

```diff
@@ -0,0 +1,71 @@
+import type { Options as SWCOptions } from '@swc/core'
+import type { CompilerOptions } from 'typescript'
+
+import { resolve } from 'node:path'
+import { parseJsonFile } from '../load-jsconfig'
+
+export function resolveSWCOptions(cwd: string): SWCOptions {
+  const tsConfig = lazilyGetTSConfig(cwd)
+
+  return {
+    jsc: {
+      parser: {
+        syntax: 'typescript',
+      },
+      paths: tsConfig?.compilerOptions?.paths,
+      // SWC requires `baseUrl` to be passed when `paths` are used.
+      // Also, `baseUrl` must be absolute.
+      baseUrl: resolve(cwd, tsConfig?.compilerOptions?.baseUrl ?? ''),
+      experimental: {
+        keepImportAttributes: true,
+        emitAssertForImportAttributes: true,
+      },
+    },
+    env: {
+      targets: {
+        node: process.versions.node,
+      },
+    },
+  } satisfies SWCOptions
+}
+
+// Since we only need "paths" and "baseUrl" from tsconfig for now,
+// we lazily look for tsconfig.json at cwd. Does not cover edge cases
+// like "extends" or even the case where tsconfig.json does not exist.
```

Comment:
I predict that this is going to be a problem. We should warn if we detect an `extends` and add a TODO to leverage tsconfig resolution from typescript itself. I know that `typescript` has methods to load the full tsconfig.

Reply:
Got it! Here is some context as to why I made it lazy:

At first, I did get the full tsconfig with `typescript`, but then we need to validate if it's installed, and if not installed, we cannot read it (or we can fall back to JSONC parse, but `extends` is still a problem). Those will be handled later at `verifyTypeScriptSetup`, so I tried to bring it down to the config load level, but it requires much information already and needed assumptions (e.g., `distDir`).

Reply:
Why would typescript not be installed at this point? That would be a problem since we can't correctly resolve the tsconfig so at this point we do need to ensure typescript is installed.

Reply:
Yes added TODO and warning https://github.com/vercel/next.js/pull/68365/files#diff-246dc95bde0542ecca1e1ca915bc2162e485ca66fb48986f4df57a97bb0ecd20R51